core/src/test/scala/integration/kafka/api/SaslPlainSslEndToEndAuthorizationTest.scala

```diff
@@ -16,12 +16,24 @@
   */
 package kafka.api
 
+import java.io.File
 import org.apache.kafka.common.protocol.SecurityProtocol
+import org.apache.kafka.common.config.SaslConfigs
+import kafka.utils.JaasTestUtils
 
 class SaslPlainSslEndToEndAuthorizationTest extends EndToEndAuthorizationTest {
   override protected def securityProtocol = SecurityProtocol.SASL_SSL
   override protected def kafkaClientSaslMechanism = "PLAIN"
   override protected def kafkaServerSaslMechanisms = List("PLAIN")
   override val clientPrincipal = "testuser"
   override val kafkaPrincipal = "admin"
+
+  // Use JAAS configuration properties for clients so that dynamic JAAS configuration is also tested by this set of tests
+  override protected def setJaasConfiguration(mode: SaslSetupMode, serverMechanisms: List[String], clientMechanisms: List[String],
+      serverKeytabFile: Option[File] = None, clientKeytabFile: Option[File] = None) {
+    super.setJaasConfiguration(mode, kafkaServerSaslMechanisms, List()) // create static config without client login contexts
```

Comment:
Oh, I notice now that the server mechanisms are still set via the singleton instance. As I said elsewhere, I do think that it would be nice for most of the code not to be aware of the singleton instance.